File: tests/python/cases/case_028_archive_inventory.py
"""
Task case 028: archive inventory payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_028_archive_inventory(input_data):
    # Task: archive inventory records and prepare transport-ready payload.
    task_name = 'archive_inventory'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'inventory',
        'stage': 'archive',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: closure captures payload and handler is retained in RETAINED_HANDLERS.
    def handler():
        return payload['checksum']
    RETAINED_HANDLERS.append(handler)
    return 'ok'

File: tests/nodejs/cases/case_028_archive_inventory.js
/**
 * Task case 028: archive_inventory
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case028ArchiveInventory(input) {
  // Task: archive inventory records and prepare transport-ready payload.
  const taskName = 'archive_inventory';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'inventory',
    stage: 'archive',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: closure captures payload and handler is retained in module-level handlers.
  retainedHandlers.push(() => payload.checksum);
  return 'ok';
}

module.exports = {
  case028ArchiveInventory,
};
